core/src/main/scala/org/bitcoins/core/psbt/PSBT.scala

```diff
@@ -1407,23 +1469,130 @@ case class InputPSBTMap(elements: Vector[InputPSBTRecord]) extends PSBTMap {
 
 object InputPSBTMap {
 
-  def fromUTXOSpendingInfo(
+  def finalizedFromUTXOSpendingInfo(
       spendingInfo: UTXOSpendingInfo,
-      unsignedTx: Transaction)(
+      unsignedTx: Transaction,
+      nonWitnessTxOpt: Option[Transaction])(
       implicit ec: ExecutionContext): Future[InputPSBTMap] = {
     val sigComponentF = BitcoinSigner
       .sign(spendingInfo, unsignedTx, isDummySignature = false)
 
     sigComponentF.map { sigComponent =>
+      val utxos = spendingInfo match {
+        case _: SegwitV0NativeUTXOSpendingInfo |
+            _: P2SHNestedSegwitV0UTXOSpendingInfo =>
+          Vector(WitnessUTXO(spendingInfo.output))
+        case _: RawScriptUTXOSpendingInfo | _: P2SHNoNestSpendingInfo |
+            _: UnassignedSegwitNativeUTXOSpendingInfo =>
+          nonWitnessTxOpt match {
+            case None => Vector.empty
+            case Some(nonWitnessTx) =>
+              Vector(NonWitnessOrUnknownUTXO(nonWitnessTx))
+          }
+      }
+
       val scriptSig =
         FinalizedScriptSig(sigComponent.scriptSignature)
       sigComponent.transaction match {
-        case _: BaseTransaction => InputPSBTMap(Vector(scriptSig))
+        case _: BaseTransaction => InputPSBTMap(utxos ++ Vector(scriptSig))
         case wtx: WitnessTransaction =>
           val witness = wtx.witness.witnesses(sigComponent.inputIndex.toInt)
           val scriptWitness = FinalizedScriptWitness(witness)
-          InputPSBTMap(Vector(scriptSig, scriptWitness))
+          val finalizedSigs =
+            if (witness != EmptyScriptWitness) {
+              Vector(scriptSig, scriptWitness)
+            } else {
+              Vector(scriptSig)
+            }
+          InputPSBTMap(utxos ++ finalizedSigs)
+      }
+    }
+  }
+
+  def fromUTXOSpendingInfo(
+      spendingInfo: UTXOSpendingInfo,
+      unsignedTx: Transaction,
+      nonWitnessTxOpt: Option[Transaction])(
+      implicit ec: ExecutionContext): Future[InputPSBTMap] = {
+    val sigPs = spendingInfo.signers
+      .map(signer => (signer.publicKey, Promise[ECDigitalSignature]()))
+      .toMap
+    val sigFs = {
+      val futures = sigPs.map { case (key, sigP) => (key, sigP.future) }.values
+      val future = Future.sequence(futures)
+
+      future.map(_.zip(sigPs.keys).toMap)
```

Comment:
IntelliJ is giving me highlighting errors here, it still compiles just IntelliJ thinks it doesn't.  Is it possible to fix this?

Reply:
Not really sure what to do about this, there are always a couple things IntelliJ's plugin scala seems currently unable to handle, hopefully they will fix this in a future update to the plugin